drivers/entropy/entropy_psa_crypto.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT zephyr_psa_crypto_rng
+
+#include <zephyr/drivers/entropy.h>
+#include <psa/crypto.h>
+
+/* API implementation: PSA Crypto initialization */
+static int entropy_psa_crypto_rng_init(const struct device *dev)
+{
+	psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+
+	ARG_UNUSED(dev);
+
+	status = psa_crypto_init();
+	if (status != PSA_SUCCESS) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/* API implementation: get_entropy */
+static int entropy_psa_crypto_rng_get_entropy(const struct device *dev,
+					      uint8_t *buffer, uint16_t length)
+{
+	psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+
+	ARG_UNUSED(dev);
+
+	status = psa_generate_random(buffer, length);
+	if (status != PSA_SUCCESS) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/* API implementation: get_entropy_isr */
+static int entropy_psa_crypto_rng_get_entropy_isr(const struct device *dev,
+						  uint8_t *buffer,
+						  uint16_t length,
+						  uint32_t flags)
+{
+	ARG_UNUSED(flags);
+
+	/* No specific handling in case of running from ISR, just call standard API */
+	return entropy_psa_crypto_rng_get_entropy(dev, buffer, length);
```

Comment:
This looks like a problem to me. Calling into the PSA API from interrupt context shouldn't be done IMO.

When PSA API is exposed by TF-M then this is not possible, since there is a mutex on the TF-M interface.
If you enable CONFIG_ASSERT and call this function from and interrupt you should see an assert happen.